core/src/test/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandlerTest.java

```diff
@@ -0,0 +1,139 @@
+import org.mockito.Spy;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+
+class ExceptionReportingServerErrorHandlerTest {
+
+    @Spy
+    final ListAppender<ILoggingEvent> logAppender = new ListAppender<>();
+    final Logger errorHandlerLogger =
+            (Logger) LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+    private static final long unLoggedExceptionReportInterval = 1;
+    private static final long awaitIntervalForTest = 2;
+
+    @BeforeEach
+    public void attachAppender() {
+        logAppender.start();
+        errorHandlerLogger.addAppender(logAppender);
+    }
+
+    @AfterEach
+    public void detachAppender() {
+        errorHandlerLogger.detachAppender(logAppender);
+        logAppender.list.clear();
+    }
+
+    @Test
+    void testServerBuilderConfiguration() {
+        final Server server1 = Server.builder()
+                                     .service("/", (ctx, req) -> HttpResponse.of(HttpStatus.OK))
+                                     .build();
+        assertThat(server1.config().unLoggedExceptionReportInterval().isZero()).isFalse();
+        assertThat(server1.config().unLoggedExceptionReportInterval().getSeconds()).isEqualTo(60);
+
+        final Server server2 = Server.builder()
+                                     .service("/", (ctx, req) -> HttpResponse.of(HttpStatus.OK))
+                                     .unLoggedExceptionReportInterval(Duration.ofSeconds(120))
+                                     .build();
+        assertThat(server2.config().unLoggedExceptionReportInterval().isZero()).isFalse();
+        assertThat(server2.config().unLoggedExceptionReportInterval().getSeconds()).isEqualTo(120);
+
+        final Server server3 = Server.builder()
+                                     .service("/", (ctx, req) -> HttpResponse.of(HttpStatus.OK))
+                                     .unLoggedExceptionReportInterval(Duration.ZERO)
+                                     .build();
+        assertThat(server3.config().unLoggedExceptionReportInterval().isZero()).isTrue();
+    }
+
+    @Test
+    void testExceptionIsLogged() throws Exception {
+        final Server server = Server.builder()
+                                    .service("/hello", (ctx, req) -> {
+                                        throw new IllegalArgumentException("test");
+                                    })
+                                    .unLoggedExceptionReportInterval(
+                                            Duration.ofSeconds(unLoggedExceptionReportInterval))
+                                    .build();
+        server.start().join();
```

Comment:
You have to stop the server you started at the end.
```
  try {
    server.start().join()
    ...
  } finally {
    server.stop();
  }
```

Please fix all tests that don't stop/close the server it created.

Reply:
All fixed !